### PR TITLE
🐛 Stream test bug fix

### PR DESCRIPTION
**Pull request by prashantgupta24**
**What this PR does / why we need it**:

This PR adds:
- A fixture for the `fastapi.testclient`
- Skips testing streaming cases with FastAPI's testclient, pending resolution of https://github.com/tiangolo/fastapi/discussions/10518
- Instead, test streaming tests against an actual server.

fix https://github.com/caikit/caikit/issues/540

**Special notes for your reviewer**:
- Following suggestions from:
-  https://github.com/encode/starlette/issues/1315#issuecomment-1166600293 
- https://github.com/sysid/sse-starlette/issues/68#issuecomment-1678266759

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: tests/runtime/http_server/test_http_server.py [hunk 1]
```diff
@@ -457,6 +457,34 @@ def test_inference_streaming_sample_module(sample_task_model_id, runtime_http_se
         )
 
 
+def test_inference_streaming_sample_module_another_test(
```
Comment (prashantgupta24) [inline tests/runtime/http_server/test_http_server.py:460]: This is the exact same test as the `test_inference_streaming_sample_module`, but for some reason, this test hands indefinitely if they are both run together

Comment (prashantgupta24) [reply tests/runtime/http_server/test_http_server.py:460]: FYI the grpc test cases run perfectly fine, from my initial investigation, it seems like a bug with asyncio or sse, both of which target the http server

File: tests/runtime/http_server/test_http_server.py [hunk 2]
```diff
@@ -43,6 +43,15 @@
     runtime_http_test_server,
 )
 
+## Fixtures #####################################################################
+
+
+@pytest.fixture(scope="session")
```
Comment (gabe-l-hart) [inline tests/runtime/http_server/test_http_server.py:49]: Why is this session scoped? I'm wondering if this is hiding the real issue of concurrent requests with multiple connections?

**Comment by joerunde** [comment]
> ... so that it's used within a context manager.

But all the tests shown already were using it within a context manager? 😕😕😕 

It looks like what this is doing is really moving to a single `TestClient` instead of having one per test. I'm totally down for this anyway since it's much DRYer but still confused about the root cause here.

File: tests/runtime/http_server/test_http_server.py (hunk 2 above)
Comment (prashantgupta24) [reply tests/runtime/http_server/test_http_server.py:49]: Yeah without the session scope it's basically the same thing we had before, and it fails on the second stream test

**Comment by prashantgupta24** [comment]
@joerunde yeah the only difference was the session scoping of the fixture that was making it work. It doesn't work without that unfortunately :(

**Review by joerunde — APPROVED** [review]
Honestly, I'm here for the test code simplification, the fixture for the client is really nice. Let us know if you find out any more details about why this thing hangs!

**Comment by prashantgupta24** [comment]
@joerunde Now I'm thinking making the fixture session-scoped is not the right thing to do. Adding another test that pings an actual running server and using that along with either a function scoped or a session scoped test fixture doesn't seem to work :( 

My gut feeling is that `TestClient` and `EventSourceResponse` don't seem to work well together. If I don't use `TestClient`, then running any number of tests against the stream endpoint seem to work.

File: tests/runtime/http_server/test_http_server.py (hunk 2 above)
Comment (gabe-l-hart) [reply tests/runtime/http_server/test_http_server.py:49]: Hm, ok, that actually seems like this doesn't really solve the problem then, right? Is this a property of the `TestClient` itself or rather a property of the server app being connected to? If it's the former, this is fine as the "problem" lies in the testing infrastructure, but if it's the later, this is just hiding that and is probably not the right fix.

Comment (prashantgupta24) [reply tests/runtime/http_server/test_http_server.py]: Removing session scoping, I agree it's not the right thing to do in this case.

**Pull request merged**